modules/core/src/controllers/orthographic-controller.js

```diff
@@ -105,25 +108,82 @@ class OrthographicState extends ViewState {
     return this;
   }
 
+  // Calculates new zoom
+  _calculateNewZoom({scale, startZoom}) {
+    const {maxZoom, minZoom} = this._viewportProps;
+    const zoom = startZoom + Math.log2(scale);
+    return clamp(zoom, minZoom, maxZoom);
+  }
+
+  /**
+   * Start zooming
+   * @param {[Number, Number]} pos - position on screen where the center is
+   */
+  zoomStart({pos}) {
+    return this._getUpdatedState({
+      startZoomPosition: pos,
+      startZoom: this._viewportProps.zoom
+    });
+  }
+
   /**
    * Zoom
    * @param {Number} scale - a number between [0, 1] specifying the accumulated
    *   relative scale.
    * @param {[number, number]} pos - current mouse cursor screen position
    */
-  zoom({pos, scale}) {
-    const {zoom, width, height, offset, minZoom, maxZoom} = this._viewportProps;
-    const newZoom = clamp(zoom * scale, minZoom, maxZoom);
+  zoom({pos, startPos, scale}) {
+    const {zoom, width, height, offset} = this._viewportProps;
+    let {startZoom, startZoomPosition} = this._interactiveState;
+    if (!Number.isFinite(startZoom)) {
+      // We have two modes of zoom:
+      // scroll zoom that are discrete events (transform from the current zoom level),
+      // and pinch zoom that are continuous events (transform from the zoom level when
+      // pinch started).
+      // If startZoom state is defined, then use the startZoom state;
+      // otherwise assume discrete zooming
+      startZoom = this._viewportProps.zoom;
+      startZoomPosition = startPos || pos;
+    }
+
+    const newZoom = this._calculateNewZoom({scale, startZoom});
     const centerX = width / 2 - offset[0];
     const centerY = height / 2 - offset[1];
-    const dX = (pos[0] - centerX) * (zoom / newZoom - 1);
-    const dY = (pos[1] - centerY) * (zoom / newZoom - 1);
+    const dX = (startZoomPosition[0] - centerX) * (zoom / newZoom - 1);
```

Comment:
- Why use `startZoomPosition` instead of `pos`?
- This is still incorrect. If `zoom` is logarithmic you need to use `scale` and `newScale` here.

Reply:
![orthographic](https://user-images.githubusercontent.com/2059298/49620883-c10bca80-f977-11e8-835b-53618c84a275.gif)